reladomographql/src/main/java/com/gs/reladomo/graphql/AggregateQueryFetcher.java

```diff
@@ -0,0 +1,168 @@
+/*
+ Copyright 2019 Goldman Sachs.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+
+package com.gs.reladomo.graphql;
+
+import com.gs.fw.common.mithra.AggregateData;
+import com.gs.fw.common.mithra.AggregateList;
+import com.gs.fw.common.mithra.MithraAggregateAttribute;
+import com.gs.fw.common.mithra.attribute.Attribute;
+import com.gs.fw.common.mithra.attribute.NumericAttribute;
+import com.gs.fw.common.mithra.finder.Operation;
+import com.gs.fw.common.mithra.finder.RelatedFinder;
+import graphql.language.Field;
+import graphql.language.Selection;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class AggregateQueryFetcher implements DataFetcher
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger (AggregateQueryFetcher.class);
+
+    private final RelatedFinder finder;
+    protected static final Map<String, Function<NumericAttribute, MithraAggregateAttribute>> AGGREGATE_METHODS = new HashMap<> ();
+
+    static
+    {
+        AGGREGATE_METHODS.put ("sumOf", NumericAttribute::sum);
+        AGGREGATE_METHODS.put ("avgOf", NumericAttribute::avg);
+        AGGREGATE_METHODS.put ("minOf", NumericAttribute::min);
+        AGGREGATE_METHODS.put ("maxOf", NumericAttribute::max);
+        AGGREGATE_METHODS.put ("standardDeviationSampleOf", NumericAttribute::standardDeviationSample);
+        AGGREGATE_METHODS.put ("standardDeviationPopulationOf", NumericAttribute::standardDeviationPopulation);
+        AGGREGATE_METHODS.put ("varianceSampleOf", NumericAttribute::varianceSample);
+        AGGREGATE_METHODS.put ("variancePopulationOf", NumericAttribute::variancePopulation);
+    }
+
+    public AggregateQueryFetcher (RelatedFinder finder)
+    {
+        this.finder = finder;
+    }
+
+    @Override
+    public Object get (DataFetchingEnvironment dataFetchingEnvironment)
+    {
+        return this.getResult (dataFetchingEnvironment.getFields ().get (0), dataFetchingEnvironment.getArguments ()); // todo: support unions with+ fields

```

Comment:
thanks for the attention. I will go over each comment and hope to address by the end of the week.

apparently the union of multiple queries would have multiple entries. Hence it would need to do get(0) union get(1) and so on. I think it has to be done with getMergedFiled() so the output comes in one JSON list.
I switched to use getField() for now. But will switch to use getMergedField with to support union.